src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/PackageSourceValidator.cs

```diff
@@ -0,0 +1,234 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NuGet.Configuration;
+
+namespace NuGet.PackageManagement.VisualStudio.Options
+{
+    internal static class PackageSourceValidator
+    {
+        internal static PackageSource FindExistingOrCreate(
+            string source,
+            string name,
+            bool isEnabled,
+            List<PackageSource> packageSources)
+        {
+            string trimmedSource = source?.Trim() ?? string.Empty;
```

Comment:
You do these checks, but it's not clear to me what happens if a null source/null name is passed here.

I think it's gonna throw (PackageSource constructor) when you try to create the source with a null/empty name/source

Is that handled?

Reply:
Good point! I originally had regex to validate this and show an appropriate error in the Add/Edit dialog.
However, since I removed that, now it would result in an exception from NuGet.Configuration.

I've now addressed this by throwing an ArgumentException from the PackageSourceValidator. Screenshots are added to **Known Bugs** in the PR description. Thanks!